### ISSUE TITLE
'grunt generate-config' results in incorrect copyright dates for config.js

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/chipper#763]
Whenever `grunt generate-config` or `grunt update` are run, it regenerates config.js with an incorrect copyright header.   This has happened several times during gas-properties development.  Someone has run `grunt update`, and then I've discovered that `grunt  update-copyright-dates` needs to be run because config.js is incorrect.

This is not a big deal, but checking `grunt  update-copyright-dates` is something that's part of the code-review checklist. 

Assigning to @ariel-phet to prioritize and assign.  Also labeling for developer meeting in case anyone has ideas for a quick fix.

**Comment by samreid** [comment]
Would it work to add `update-copyright-dates` as a new final step in `grunt update`?

**Comment by pixelzoom** [comment]
That would probably be fine, as long as we don't mind the fact that `grunt update-copyright-dates` will hit files that aren't relevant to `grunt generate-config` and `grunt update`.

**Comment by zepumph** [comment]
We think that update-copy-dates, along with generating the third part report, should be done on bayes automatically. We can create a script in perennial/bin to run this daily grunt work. See https://github.com/phetsims/perennial/issues/141 for that work. Is this ready to be closed?

**Comment by pixelzoom** [comment]
👍

**Comment by zepumph** [comment]
I'm realizing the above plan didn't fix the bug that created this issue. The above commit fixes this, as well as 
https://github.com/phetsims/perennial/issues/120

**Comment by zepumph** [comment]
@pixelzoom please review.

**Comment by pixelzoom** [comment]
I'm not clear on what was done, what I'm reviewing, and how I should verify.  @zepumph please clarify.

**Comment by zepumph** [comment]
@pixelzoom I realized after closing that  https://github.com/phetsims/perennial/issues/141 does not solve this problem, so I committed a fix in https://github.com/phetsims/chipper/commit/0792ef142f89471a151d6a1b197b6e4686662bea to the original problem of this issue.

The issue is the `generate-config` wants to run `updateCopyrightDate` on just the config file when the config file is being overwritten, but `update-copyright-dates` errors out within `generate-config` when there is no config file already checked in (i.e. for `create-sim` task see https://github.com/phetsims/perennial/issues/120). Please review the linked commit, and note fix as a solution to both this issue and https://github.com/phetsims/perennial/issues/120. 

Here are some cases that should generate a config that has the right copyright date:
* run `grunt update` in a sim. Note that now `updateCopyrightDate` is now run for the `generate-config` grunt task, fixing the filled in template.
* run `createSim` and notice that it has the correct copyright date for the config file (only one number because the copyright update script wasn't run. This copyright date was taken directly from the config template.
* Try something else weird like deleting your local copy config file in a checked in repo, then running grunt update. Things should still look the same as the first bullet.

Let me know if you have questions.

**Comment by pixelzoom** [comment]
Slack:

Chris Malley [3:16 PM]
Questions about https://github.com/phetsims/chipper/issues/763…
(1) If I run `grunt update` will it potentially modify copyright dates in files other than config.js?
(2) Are files with modified copyright dates committed and pushed?


Michael Kauzmann [3:16 PM]
1. no
2. no

Michael Kauzmann [3:17 PM]
even the `update-copyright-dates` grunt task doesn't commit and push
I had to add that manually into the automated task I wrote last week.
